scipy/_lib/tests/test__util.py

```diff
@@ -250,13 +251,15 @@ def test_rng_integers():
 
 class TestValidateInt:
 
-    def test_validate_int(self):
-        n = _validate_int(4, 'n')
+    @pytest.mark.parametrize('n', [4, np.uint8(4), np.int16(4), np.array(4)])
+    def test_validate_int(self, n):
+        n = _validate_int(n, 'n')
         assert n == 4
 
-    def test_validate_int_bad1(self):
+    @pytest.mark.parametrize('n', [4.0, np.array([4]), Fraction(4, 1)])
```

Comment:
I guess I didn't notice that it was this strict. Some SciPy functions accept everything under the sun; this seems like a bit of a departure from the norm.  Thoughts on why converting to an `int` internally and checking for equality with the original is not good enough?  (This is Python, right?)